rbx_binary/src/deserializer.rs

```diff
@@ -668,7 +668,36 @@ impl<R: Read> BinaryDeserializer<R> {
                     });
                 }
             },
-            Type::BrickColor => {}
+            Type::BrickColor => match canonical_type {
+                VariantType::BrickColor => {
+                    let mut values = vec![0; type_info.referents.len()];
+                    chunk.read_interleaved_u32_array(&mut values)?;
+
+                    for (value, referent) in values.into_iter().zip(&type_info.referents) {
+                        let instance = self.instances_by_ref.get_mut(referent).unwrap();
+                        let color = BrickColor::from_number(value as u16).ok_or_else(|| {
```

Comment:
This cast won't do what we want for values that don't fit in a `u16`. I see two ways we can fix this:

1. Use `value.try_into()` (from the `std::convert::TryInto` trait), which will let us handle out-of-range values.
2. Switch `BrickColor` to be backed by a `u32`. It's not clear to me if `BrickColor` is 32 bits in the file format because it was convenient for Roblox, or if that's the intended size of a `BrickColor`.

Reply:
My guess is because it's convenient more than it being 'correct'. I don't think there's any reason for us to explicitly mirror the size in rbx-dom though.

I suppose we can also use "try_into" though. The end-goal is the same; we'll be raising an identical error.

Reply:
To be clear, using `as u16` will never raise an error. It will silently wrap and put a smaller value into `BrickColor::new`. See this playground for an example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2a8058e6e4ac5b83fdbafd0258272e7e